aequilibrae/paths/route_choice.py

```diff
@@ -174,7 +182,21 @@ def prepare(self, nodes: Union[List[int], List[Tuple[int, int]]]) -> None:
                 provided, OD pairs are taken to be all pair permutations of the list. If a list of pairs is provided
                 OD pairs are taken as is. All node IDs must be present in the compressed graph. To make a node ID
                 always appear in the compressed graph add it as a centroid. Duplicates will be dropped on execution.
+                If *None* is provided, all OD pairs with non-zero flows will be used.
         """
+        if nodes is None:
+            idx = self.matrix.index
+            if len(self.matrix.view_names) > 1:
+                m = self.matrix.matrix_view.sum(axis=2)
+            else:
+                m = self.matrix.matrix_view
+            non_zero = np.where(m > 0)
+            o = non_zero[0][non_zero[0] != non_zero[1]]
+            d = non_zero[1][non_zero[0] != non_zero[1]]
```

Comment:
maybe worth a comment here
```suggestion
            # Remove intrazonals (elements on the diagonal)
            o = non_zero[0][non_zero[0] != non_zero[1]]
            d = non_zero[1][non_zero[0] != non_zero[1]]
```

Reply:
Thanks, I've added it to my local copy on the other PR